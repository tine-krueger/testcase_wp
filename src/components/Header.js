import styled from 'styled-components/macro'
import elphi_600 from '../assets/elphi_600.jpg'

export default function Header() {

    return (
        <HeaderElphi>
            <Headline>
                <h1>Moin Moin</h1>
            </Headline>
           
        </HeaderElphi>
    )
}

const HeaderElphi =  styled.div`
    padding: 1em;    
    height: 50vh;
    background: url(${elphi_600});
    background-position: center;
    background-repeat: no-repeat;
    position: relative;

    &:before {
        content: '';
        display: block;
        background-image: radial-gradient(#3e3e3e 33%, transparent 33%);
        backdrop-filter: saturate(2);
        background-size: 3px 3px;
        position: absolute;
        top: 0;
        left: 0;
        right: 0;
        bottom: 0;

    }

`
const Headline = styled.div`
    background: rgba(0,0,0,.2);
    backdrop-filter: blur(3px);
    border-radius: 5px;
    border: 1px solid var(--secondaryFont);
    position: absolute;
    top: 50%;
    left: 29%;
    transform: translate(-19%,-50%);

    h1 {
        text-transform: uppercase;
        color: var(--secondaryFont);
        margin: .25em .5em;
    } 
`