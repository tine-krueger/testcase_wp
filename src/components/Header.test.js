import { render } from '@testing-library/react'
import 'jest-styled-components'
import Header from './Header'

describe ('Header', () => {
    it('renders correctly', () => {
        const { container } = render(
            <Header />
        )
        expect(container).toMatchSnapshot()
    })

    it('shows the correct text', () => {
        const { getByText} = render(
            <Header />
        )
        expect(getByText(/Moin Moin/i)).toBeInTheDocument()
    })
})