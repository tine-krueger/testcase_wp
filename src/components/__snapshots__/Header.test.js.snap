// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.c0 {
  padding: 1em;
  height: 50vh;
  background: url(elphi_600.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  position: relative;
}

.c1 {
  background: rgba(0,0,0,.2);
  -webkit-backdrop-filter: blur(3px);
  backdrop-filter: blur(3px);
  border: 1px solid var(--secondaryFont);
  position: absolute;
  top: 50%;
  left: 29%;
  -webkit-transform: translate(-19%,-50%);
  -ms-transform: translate(-19%,-50%);
  transform: translate(-19%,-50%);
}

.c1 h1 {
  text-transform: uppercase;
  color: var(--secondaryFont);
  margin: .25em .5em;
}

@media (min-width:768px) {
  .c0 {
    background: url(elphi_1000.jpg);
    background-position: center;
    background-repeat: no-repeat;
  }
}

@media (min-width:992px) {
  .c0 {
    background: url(elphi_1900.jpg);
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    padding: 5em;
  }
}

@media (min-width:1200px) {
  .c0 {
    padding: 10em;
  }
}

@media (min-width:768px) {
  .c1 {
    top: 50%;
    left: 50%;
    -webkit-transform: translate(-50%,-50%);
    -ms-transform: translate(-50%,-50%);
    transform: translate(-50%,-50%);
  }
}

@media (min-width:992px) {
  .c1 {
    top: 4em;
    left: 9em;
  }
}

@media (min-width:1200px) {
  .c1 {
    top: 6em;
    left: 13em;
  }

  .c1 h1 {
    font-size: 2.8rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1>
        Moin Moin
      </h1>
    </div>
  </div>
</div>
`;
